src/process_quantiles.slurm

```diff
@@ -9,11 +9,11 @@
 #SBATCH --mail-user=mhines@usgs.gov
```

Comment:
Along the same lines, you may be able to reduce the --mem requirement if you decide to cut down on how many NetCDF files are open at once (see previous comments).

Reply:
@mhines-usgs ☝️

Reply:
so 110 tasks * ~5 minutes = ~550 minutes = ~9.16666666 hours?  I went with 9 hours 🤞 but it is hitting this 'reason' for not starting on Yeti

When I do the sbatch for this, I'm getting this in my `sidle` response
```
Queue for mhines
JOBID         PARTITION         NAME        TIME ST CPUS    TIME_LEFT  NODELIST(REASON)
4774976_[1-11    UV,normal process_quan        0:00    PD    1      9:00:00  (BadConstraints)
```
here's my full slurm
```
[mhines@yeti-login20 wbeep-processing] cat src/process_quantiles.slurm
#!/bin/bash
#SBATCH --job-name=process_quantiles     # name that you chose
#SBATCH -c 1                   # number of cores per task
#SBATCH -p UV,normal              # the partition you want to use, for this case prod is best
#SBATCH -A iidd                 # your account
#SBATCH --time=09:00:00        # time at which the process will be cancelled if unfinished
#SBATCH --mem=200G
#SBATCH --mail-type=ALL
#SBATCH --mail-user=mhines@usgs.gov
#SBATCH -o /logs/%A_%a-%j.log            # log file for each jobid (can insert %A_%a for each array id task if needed)
#SBATCH --export=ALL
#SBATCH --array=1-110 # 109,951 HRUs / 1000 HRUs per task = 110 tasks

mkdir -p logs
mkdir -p grouped_quantiles

module load R/3.5.1-gcc7.1.0

srun Rscript src/process_quantiles.R
```

Reply:
I think `--time` refers to the expected time per array element, not for the whole batch - so could safely be more like `00:10:00`. is it possible that `normal` requires a max of `07:00:00` or something like that? I forget what the limits are. But regardless, the shorter time should work.

Reply:
Hmm, but that must not be the `BadConstraints` because "Yeti has time limits on each partition as follows: normal = 7 days, UV = 3 days, large = 3 days, long = 30 days."

Reply:
Maybe it doesn't like `/logs/%A_%a-%j.log` and would prefer `./logs/%A_%a-%j.log`?

Reply:
👍 ill try adjusting both of these.. thanks

Reply:
finally got things to start running, I tested a lot of different values for the SBATCH options for mem before getting it to finally go... running now and working so far 🤞 

I did originally hit an error even with this config about a problem loading the ncdf4 library from David's workspace, so had to switch my workspace over to my own temporarily in my .Rprofile (have been using his for another task!)
```
#!/bin/bash
#SBATCH --job-name=process_quantiles     # name that you chose
#SBATCH -p normal,UV              # the partition you want to use, for this case prod is best
#SBATCH -A iidd                 # your account
#SBATCH -o logs/slurm-%A_%a-%j.out
#SBATCH --time=00:10:00        # time at which the process will be cancelled if unfinished
#SBATCH --mem=10GB
#SBATCH --mail-type=ALL
#SBATCH --mail-user=mhines@usgs.gov
#SBATCH --export=ALL
#SBATCH --array=1-110 # 109,951 HRUs / 1000 HRUs per task = 110 tasks
```

Reply:
Although it just invalidated a bunch of the tasks after they hit the 10 minute time limit. I must have something not quite right if it's taking that much longer?

Reply:
I wonder if data.table is doing something parallel on Lindsay's computer - it might then be slower when you just have one core per task (`-c 1`), which could explain the difference. But I'd expect that when you have 110 tasks going at once, `-c 1` is still a fine option. So maybe just bump up the expected time to...I dunno, 30 minutes? 60?

Reply:
Did it invalidate all of them or does it look like some completed and have files in `quantiles_grouped/`? It may not have had 110 cores available to run them all simultaneously ?

Reply:
Hmm well I bumped things up to 30 minutes and i'm 17 minutes in on a few, and those have just read in the second of 6 nc files according to the logs. I think we may need to do more tuning on this? Or else maybe the files from Steve are a lot larger than expected? They're ~ 10.4gb each

Reply:
Did some of the tasks complete and others time out? If so, maybe 20 minutes would be enough.

Reply:
There's a lag in this thread =)

Reply:
How about bumping up `-c` to 4 or 8?

Reply:
And yeah, comparing new to old file sizes seems wise.

Reply:
The netcdf files I am using locally are 11.6 GB each so I don't think it is the file size. And you definitely pulled down these changed files to use, right?

Reply:
yep, correct, these are the files Steve put on FTP - I put them up on Yeti here - /cxfs/projects/usgs/water/iidd/data-sci/historical_2019102.zip and then unzipped them into my workspace.